### ISSUE TITLE
[BUG] Alias of external package makes independent‑modules fail allowExternal

### ISSUE THREAD

**Issue by nicolasventer** [Igorkowalski94/eslint-plugin-project-structure#29]
Hello, the issue can be reproduced with the steps below.

# **Environment**

Here are the version used:
- eslint-plugin-project-structure: `3.13.0`
- eslint: `9.17.0`
- vite: `6.0.5`
- bun: `1.1.37`
- preact: `10.25.4`

# **Describe the bug**

When making alias of react to preact/compat in tsconfig, `independent‑modules` raise issue when importing **external modules**. *(Even sometimes there is issue with internal modules.)*

# **Expected behavior**

The `independent‑modules` should not raise issue when the import is allowed, either external or internal.

# **Additional context**

Here are the steps to reproduce:

## 1. Create Project

```sh
bunx create vite
```

Select React. Select Typescript.

## 2. Install dependencies

```sh
cd vite-project
bun install
bun install -D eslint-plugin-project-structure
```

## 3. Install all dependencies (optional)

```sh
bun install preact
```

change `package.json` (see attachment)  
short content:
```json
		"react": "npm:@preact/compat",
		"react-dom": "npm:@preact/compat",
```

```sh
bun install
```

## 4. Setup Independent Modules

setup `independentModule.mjs` (see attachment)  
short content:
```ts
		{
			name: "Root",
			pattern: ["*", "src/*"],
			allowImportsFrom: ["*", "src/*", "src/assets/*"],
		},
```

setup `eslint.config.js` (see attachment)

## 5. Check that there is no issue

```sh
bun run lint
```

## 6. Configure tsconfig (this step cause the issue)

delete `tsconfig.node.json` and `tsconfig.json`  
rename `tsconfig.app.json` to `tsconfig.json`  
add this in compilerOptions of `tsconfig.json` (see attachment)  
short content:
```json
		"paths": {
			"react": ["./node_modules/preact/compat/"],
			"react/jsx-runtime": ["./node_modules/preact/jsx-runtime"],
			"react-dom": ["./node_modules/preact/compat/"],
			"react-dom/*": ["./node_modules/preact/compat/*"]
		},
```

## 7. Check that there is an issue

```sh
bun run lint
```

Output:
```
$ eslint .

...\src\App.tsx
  1:1  error  🔥 This import is not allowed in the module 'Root'. 🔥  project-structure/independent-modules
  2:1  error  🔥 This import is not allowed in the module 'Root'. 🔥  project-structure/independent-modules

...\src\main.tsx
  1:1  error  🔥 This import is not allowed in the module 'Root'. 🔥  project-structure/independent-modules
  2:1  error  🔥 This import is not allowed in the module 'Root'. 🔥  project-structure/independent-modules

...\vite.config.ts
  2:1  error  🔥 This import is not allowed in the module 'Root'. 🔥  project-structure/independent-modules

✖ 5 problems (5 errors, 0 warnings)

error: script "lint" exited with code 1
```

<details>
<summary>Optional try</summary>

even changing `allowImportsFrom` to `allowImportsFrom: ["*", "src/*", "src/assets/**", "node_modules/**"],` is not enough.

```sh
bun run lint
```

Output:
```
$ eslint .

...\src\App.tsx
  2:1  error  🔥 This import is not allowed in the module 'Root'. 🔥  project-structure/independent-modules

...\vite.config.ts
  2:1  error  🔥 This import is not allowed in the module 'Root'. 🔥  project-structure/independent-modules

✖ 2 problems (2 errors, 0 warnings)

error: script "lint" exited with code 1
```

</details>

___

Can you please take a look at this issue? Thank you.

***By the way, your plugin is awesome!***

[independent-modules-issue-attachements.zip](https://github.com/user-attachments/files/18300230/independent-modules-issue-attachements.zip)

[independent-modules-issue-project.zip](https://github.com/user-attachments/files/18300233/independent-modules-issue-project.zip)

**Comment by Igorkowalski94** [comment]
Hi @nicolasventer, can you show me the error message for a specific import with the flag ```{ "debugMode": true }``` enabled?

**Comment by Igorkowalski94** [comment]
In general, if you create an alias for external imports, you must manually specify them later in `allowImportsFrom`.

For example, like this:

```mjs
	{
            name: "Root",
	    pattern: ["*", "src/*"],
            allowImportsFrom: ["*", "src/*", "src/assets/*", "**/node_modules/preact/**"],
        },
```

**Comment by Igorkowalski94** [comment]
In summary, if you added external imports as aliases, the library will no longer recognize them as external imports, and you will need to allow or disallow them yourself in `allowImportsFrom`. For now, I’m unlikely to change this behavior or add additional checks for aliases, as this is a rather rare case, and the workaround is fairly straightforward.

Let me know if this workaround is okay for you, and close the issue if everything works as it should.

**Comment by nicolasventer** [comment]
*(Sorry, I've just seen your reploy. Thanks for the reactivity.)*

Here is the result with `debugMode: true`:

```
eslint .

C:\Users\thede\Desktop\Projects\independant-modules-issue\src\App.tsx
  2:1  error  🔥 This import is not allowed in the module 'Root'. 🔥

File path   = "src/App.tsx"
Import path = "src/assets/node_modules/preact/compat/.svg"
{family}    = "NO_FAMILY"
{dirname}   = "src"

allowImportsFrom:
"*"
"src/*"
"src/assets/*"
"**/node_modules/preact/**"
"**/node_modules/react/**"
"**/node_modules/**"

  project-structure/independent-modules

C:\Users\thede\Desktop\Projects\independant-modules-issue\vite.config.ts
  2:1  error  🔥 This import is not allowed in the module 'Root'. 🔥

File path   = "vite.config.ts"
Import path = "@vitejs/plugin-node_modules/preact/compat/"
{family}    = "NO_FAMILY"
{dirname}   = "\."

allowImportsFrom:
"*"
"src/*"
"src/assets/*"
"**/node_modules/preact/**"
"**/node_modules/react/**"
"**/node_modules/**"
```

With debug mode, I have added many values in allowImportsFrom to have no  issue. Here is the values:
```ts
		{
			name: "Root",
			pattern: ["*", "src/*"],
			allowImportsFrom: [
				"*",
				"src/*",
				"src/assets/*",
				"node_modules/**",
				"@vitejs/plugin-node_modules/**",
				"src/assets/node_modules/**",
				"src/assets/node_modules/preact/compat/.svg", // this does not work without this line
			],
		},
```

Do you approve this workaround ?

**Comment by nicolasventer** [comment]
This values works. I'm fine with this. Thank you for the help.

**Comment by Igorkowalski94** [comment]
Thank you for reporting the issue. In the future, I will think of a better solution for such cases.

**Comment by nicolasventer** [comment]
> ```js
> {
> 			name: "Root",
> 			pattern: ["*", "src/*"],
> 			allowImportsFrom: [
> 				"*",
> 				"src/*",
> 				"src/assets/*",
> 
>                                  // It should rather cover all cases, considering the errors you pasted.
> 				"**/preact/**,
>                               
>                                 // It should work this way if you designate a dot as a symbol.
> 				"src/assets/node_modules/preact/compat/\\.svg", 
> 
> 			],
> },
> ```

Sorry for bothering you again, but I tried this workaround in a bigger project with more dependencies, and it seems that the weird behaviour with the `react.svg` is that the first "word" of the filename is truncated.

In debug mode with my other project I have this issue:
```
  5:1  error  🔥 This import is not allowed in the module 'Routes'. 🔥

File path   = "src/routes/index.lazy.tsx"
Import path = "node_modules/preact/compat/-hot-toast"
{family}    = "NO_FAMILY"
{dirname}   = "src/routes"
```
for the import of `react-hot-toast`

I think there is something to see on this.
